src/plugins/plugin-manager.ts

```diff
@@ -45,23 +51,32 @@ export class PluginManager {
     'performance',
     'browserconsole',
   ];
-  constructor(private driver: Driver) {}
+  constructor(private driver: Driver | APIDriver) {}
 
   register(type: PluginType, options: PluginOptions) {
     let instance: Plugin;
-    switch (type) {
-      case 'network':
-        instance = new NetworkManager(this.driver);
-        break;
-      case 'trace':
-        instance = new Tracing(this.driver, options);
-        break;
-      case 'performance':
-        instance = new PerformanceManager(this.driver);
-        break;
-      case 'browserconsole':
-        instance = new BrowserConsole(this.driver);
-        break;
+
+    if ('context' in this.driver) {
```

Comment:
Plugin Manager should be transparent to the underlying journey that was run whether its browser/api based. There will be always a chance when browser/api based journeys coexisting. I would handle these and make them respective plugins transparent.